src/util/nchan_accumulator.c

```diff
@@ -0,0 +1,198 @@
+#include <math.h>
+#include <nchan_module.h>
+
+#include "nchan_accumulator.h"
+
+// based on work done for Queris (c) 2011-2017 Leo Ponomarev
+// https://github.com/slact/queris/blob/master/lib/queris/indices.rb#L661-L712
+// licensed under the MIT license, same as Nchan
+
+int nchan_accumulator_init(nchan_accumulator_t *acc, nchan_accumulator_type_t type, double halflife) {
+  switch(type) {
+    case ACCUMULATOR_EXP_DECAY_FLOAT:
+      acc->data.ed_float.value = 0;
+      acc->data.ed_float.weight = 0;
+      acc->data.ed_float.lambda = (double)1 / halflife;
```

Comment:
can `halflife` be zero here (or close to it) ?

Reply:
Techincally a halflife of 0 would be undefined behavior. 

In practice, lambda would be set to NaN, and the SIGABRT would happen later.

Reply:
We don't use this code now, but may in the future. Fixed in 81346574.